Source/UrlRequest_Windows.cpp

```diff
@@ -172,37 +168,57 @@ namespace UrlLib
 
         gsl::span<const std::byte> ResponseBuffer() const
         {
-            if (!m_responseBuffer)
+            if ((m_uri.SchemeName() == L"app" || m_uri.SchemeName() == L"file") && m_fileResponseBuffer.size() > 0)
             {
-                return {};
+                return {(std::byte*)m_fileResponseBuffer.data(), gsl::narrow_cast<std::size_t>(m_fileResponseBuffer.size())};
             }
-
-            std::byte* bytes;
-            auto bufferByteAccess = m_responseBuffer.as<::Windows::Storage::Streams::IBufferByteAccess>();
-            winrt::check_hresult(bufferByteAccess->Buffer(reinterpret_cast<byte**>(&bytes)));
-            return {bytes, gsl::narrow_cast<std::size_t>(m_responseBuffer.Length())};
+            else if (m_httpResponseBuffer)
+            {
+                std::byte* bytes{nullptr};
+                auto bufferByteAccess = m_httpResponseBuffer.as<::Windows::Storage::Streams::IBufferByteAccess>();
+                winrt::check_hresult(bufferByteAccess->Buffer(reinterpret_cast<byte**>(&bytes)));
+                return {bytes, gsl::narrow_cast<std::size_t>(m_httpResponseBuffer.Length())};
+            }
+            return {};
         }
 
     private:
-        arcana::task<void, std::exception_ptr> LoadFileAsync(Storage::StorageFile file)
+        arcana::task<void, std::exception_ptr> LoadFileAsync(const std::wstring& path)
         {
             switch (m_responseType)
             {
                 case UrlResponseType::String:
                 {
-                    return arcana::create_task<std::exception_ptr>(Storage::FileIO::ReadTextAsync(file))
-                        .then(arcana::inline_scheduler, m_cancellationSource, [this](winrt::hstring text) {
-                            m_responseString = winrt::to_string(text);
-                            m_statusCode = UrlStatusCode::Ok;
-                        });
+                    return arcana::make_task(arcana::threadpool_scheduler, m_cancellationSource, [this, path] {
+                        std::ifstream file(path);
+                        if (!file.good())
+                        {
+                            std::stringstream msg;
+                            msg << "Failed to read file " << path.c_str();
+                            throw std::runtime_error{msg.str()};
+                        }
+
+                        std::stringstream ss;
+                        ss << file.rdbuf();
+
+                        m_responseString = ss.str();
+                        m_statusCode = UrlStatusCode::Ok;
```

Comment:
I'm worried about file encoding here. Does this code handle that at all? Maybe we can punt this for later since I don't think the existing code worked correctly either.

Reply:
I vote we punt it.

Reply:
Let's incorporate into this issue: https://github.com/BabylonJS/BabylonNative/issues/228

Reply:
Updated linked issue description to include this.